core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -393,6 +396,11 @@ public Configuration getConf() {
     return conf;
   }
 
+  @Override
+  protected Map<String, String> properties() {
+    return catalogProperties;
```

Comment:
Nit: For here, _if it will remove the not-initialized warning from `catalogProperties`, we often do the following:

```java
return catalogProperies != null ? ImmutableMap.of() : catalogProperties;
```